### PR TITLE
More tracing spans, fix spans in tokio::spawn()

### PR DESCRIPTION
**Pull request by Xiretza**

### CODE REVIEW

File: src/service/presence/mod.rs [hunk 1]
```diff
@@ -187,30 +194,31 @@ impl Service {
...
-							presence_timers.push(presence_timer(user_id, timeout));
-						}
-						Err(e) => {
-							// generally shouldn't happen
-							error!("Failed to receive presence timer through channel: {e}");
-						}
+	#[tracing::instrument(skip_all, name = "presence_loop")]
+	async fn handler_inner(
+		&self, receiver: &MutexGuard<'_, loole::Receiver<(OwnedUserId, Duration)>>,
+		presence_timers: &mut FuturesUnordered<BoxFuture<'_, OwnedUserId>>,
```
Comment (Xiretza) [inline src/service/presence/mod.rs:200]: Not too happy about this, but haven't found a nice way to create a nameable future that returns a value after some time.

**Comment by Xiretza** [comment]
I think [the span on `router::request_spawn()`](https://github.com/girlbossceo/conduwuit/blob/99d98efeb187bcac21856dedbdfbb7d458a6a176/src/router/mod.rs#L70) can also be removed, it never emits any events, but I'd like a second opinion on that.

**Comment by jevolk** [comment]
> I think [the span on `router::request_spawn()`](https://github.com/girlbossceo/conduwuit/blob/99d98efeb187bcac21856dedbdfbb7d458a6a176/src/router/mod.rs#L70) can also be removed, it never emits any events, but I'd like a second opinion on that.

I feel like I've seen it once during internal server errors from other parts of the tower stack outside of the request handler. Not sure though. Is there any advantage to removing it?

**Comment by Xiretza** [comment]
The advantage would be not having two root spans for every incoming request, one of which is (almost always?) useless.

**Comment by jevolk** [comment]
> The advantage would be not having two root spans for every incoming request, one of which is (almost always?) useless.

Does that affect or clutter something in jaeger? I don't want to lose coverage over a portion of the stack if there isn't problem from it.

**Comment by Xiretza** [comment]
Yes:

image

`router:` are the spans containing actual information, `spawn` is always an empty span.

The only thing `request_spawn` does outside of the spawned task (which gets its own root span in `tracing_span()`) is load an atomic and call `Next::run` (the sync part, the future itself is evaluated inside the spawned task). Neither of those can emit events.

**Comment by Xiretza** [comment]
I'm honestly not sure if `request_spawn()` is required in the first place, axum spawns a separate thread per request anyway.

**Comment by Xiretza** [comment]
Ahh, I guess the axum task gets cancelled when the client disconnects, that's why we spawn our own: https://github.com/girlbossceo/conduwuit/commit/da3871f39a3a0a7b5b34e2110e41717c33754a73

**Comment by Xiretza** [comment]
Alright, spans fixed for good now. The spans in the long-running tasks were actually still too wide and included the sleep/receive/etc, which created absurdly long span durations. With these changes, filtering for span durations actually produces useful results.

**Pull request closed without merging**